pkg/artifactory/resource_artifactory_single_replication_config.go

```diff
@@ -211,6 +211,10 @@ func resourceSingleReplicationConfigExists(d *schema.ResourceData, m interface{}
 	replicationName := d.Id()
 	replicationConfig, resp, err := c.V1.Artifacts.GetRepositoryReplicationConfig(context.Background(), replicationName)
 
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.